wfe2/wfe.go

```diff
@@ -592,7 +592,21 @@ func (wfe *WebFrontEndImpl) Nonce(
 }
 
 // sendError wraps web.SendError
-func (wfe *WebFrontEndImpl) sendError(response http.ResponseWriter, logEvent *web.RequestEvent, prob *probs.ProblemDetails, ierr error) {
+func (wfe *WebFrontEndImpl) sendError(response http.ResponseWriter, logEvent *web.RequestEvent, eerr any, ierr error) {
+	// TODO(#4980): Simplify this function to only take a single error argument,
+	// and use web.ProblemDetailsForError to extract the corresponding prob from
+	// that. For now, though, the third argument has to be `any` so that it can
+	// be either an error or a problem, and this function can handle either one.
+	var prob *probs.ProblemDetails
+	prob, ok := eerr.(*probs.ProblemDetails)
+	if !ok {
+		err, ok := eerr.(error)
+		if !ok {
+			panic("wfe.sendError got %#v (type %T), but expected ProblemDetails or error")
+		}
+		prob = web.ProblemDetailsForError(err, "")
+	}
```

Comment:
Optional: to my eye, this would be a little a clearer as a `switch`:

```suggestion
	var prob *probs.ProblemDetails
	switch v := eerr.(type) {
	case *probs.ProblemDetails:
		prob = v
	case error:
		prob = web.ProblemDetailsForError(v, "")
	default:
		panic(fmt.Sprintf("wfe.sendError got %#v (type %T), but expected ProblemDetails or error", eerr, eerr))
	}
```

Reply:
Good call, switch is way cleaner.